src/Assembler.cpp

```diff
@@ -150,3 +150,7 @@ void Assembler::storeAssemblyTime(
     assemblerInfo->averageCpuUtilization = averageCpuUtilization;
 }
 
+void Assembler::storePeakMemoryUsage(const uint64_t peakMemoryUsage) {
```

Comment:
I would avoid arguments that are `const` and passed by value. There is some discussion on the web of why this is a questionable thing to do.

Reply:
It doesn't matter in this case. In general, any `const` directive is good for enforcing that someone doesn't accidentally modify the value (even if it is a copy).

Reply:
Of course. But the rationale for avoiding this is that the fact that the function does not modify the value of its copy is internal business of the function and should not be visible in its declaration, accessible to the caller. If you want to make it clear that the function is not modifying its copy you could use a pattern like this:

```
void f(T x)
{
    const T& xConst = x;
}
```

There is some discussion of this [here](http://www.cplusplus.com/forum/general/135731/), where somebody also suggests the possibility of using the `const` qualifier in the definition of the function but not in the declaration. That would avoid the extra complexity of the above pattern. 

But this gets a bit into philosophical differences, so feel free to write it as you please.